xarray/backends/api.py

```diff
@@ -562,12 +565,25 @@ def open_mfdataset(paths, chunks=None, concat_dim=_CONCAT_DIM_DEFAULT,
 
     if lock is None:
         lock = _default_lock(paths[0], engine)
-    datasets = [open_dataset(p, engine=engine, chunks=chunks or {}, lock=lock,
-                             **kwargs) for p in paths]
-    file_objs = [ds._file_obj for ds in datasets]
 
-    if preprocess is not None:
-        datasets = [preprocess(ds) for ds in datasets]
+    open_kwargs = dict(engine=engine, chunks=chunks or {}, lock=lock,
+                       autoclose=autoclose, **kwargs)
+
+    if parallel:
+        import dask.bag as db
+        paths_bag = db.from_sequence(paths)
+        datasets = paths_bag.map(open_dataset,
+                                            **open_kwargs).compute()
```

Comment:
What exactly does `compute()` do when called on a dask bag of xarray datasets of dask arrays? I'm assuming that only the bags get computed, but not the array values? This would be good to clarify in some comments on the code.

Reply:
That is my understanding. In other words, `delayed(open_dataset)(...).compute()` is going to return whatever `open_dataset` would have. I have not tested this thoroughly and I would be interested to hear from @mrocklin on how dask is going to treat this situation.